hack/local-up-cluster.sh

```diff
@@ -1252,9 +1252,9 @@ EOF
 }
 
 function install_cni_if_needed {
-  echo "Checking CNI Installation at /opt/cni/bin"
-  if ! command -v /opt/cni/bin/loopback &> /dev/null ; then
-    echo "CNI Installation not found at /opt/cni/bin"
+  echo "Checking CNI Installation at /opt/cni/bin and Bridge networking config file for containerd containers at '/etc/cni/net.d/10-containerd-net.conflist"
+  if ! command -v /opt/cni/bin/loopback &> /dev/null && ! command -v /etc/cni/net.d/10-containerd-net.conflist &> /dev/null; then
+    echo "CNI Installation not found at /opt/cni/bin and/or Bridge networking for containerd containers config file '10-containerd-net.conflist' not found at /etc/cni/net.d/ "
```

Comment:
On my machine, I have `/opt/cni/bin/loopback` but no `/etc/cni/net.d/10-containerd-net.conflist`. The script works for me when I invoke it like this:
```
$ cat /etc/resolv-9999.conf
nameserver 9.9.9.9

$ KUBELET_RESOLV_CONF="/etc/resolv-9999.conf" DNS_ADDON="coredns" CGROUP_DRIVER=systemd CONTAINER_RUNTIME_ENDPOINT=unix:///var/run/crio/crio.sock ALLOW_PRIVILEGED=1 ./hack/local-up-cluster.sh

```

So is my CNI installation incomplete? With your change, it would invoke `install_cni`. Perhaps that is the right thing to do because it is out-dated. But if we let local-up-cluster.sh "own" `/opt/cni` and `/etc/cni`, wouldn't it be better to have a more general mechanism for "CNI installation is up-to-date"?

Reply:
>On my machine, I have /opt/cni/bin/loopback but no /etc/cni/net.d/10-containerd-net.conflist. The script works for me when I invoke it like this:

>$ cat /etc/resolv-9999.conf
>nameserver 9.9.9.9

>$ KUBELET_RESOLV_CONF="/etc/resolv-9999.conf" DNS_ADDON="coredns" CGROUP_DRIVER=systemd >CONTAINER_RUNTIME_ENDPOINT=unix:///var/run/crio/crio.sock ALLOW_PRIVILEGED=1 ./hack/local-up-cluster.sh


Thanks for your review.
It appears that you are using `crio` where as, I was using `containerd` for runtime, in my testing and apparently with `container.d` this is a reproducible problem as I think it's expecting that file to be in place for networking setup.

>So is my CNI installation incomplete? With your change, it would invoke install_cni. Perhaps that is the right thing to do because it is out-dated. But if we let local-up-cluster.sh "own" /opt/cni and /etc/cni, wouldn't it be better to have a more general mechanism for "CNI installation is up-to-date"?

Like i mentioned above, I think your CNI installation might be complete given you are using `crio` runtime as long as your worker nodes and pods are coming up fine on your setup. I think this may be specific to `containerd` runtime setup.
It appears that script defaults to [container.d runtime](https://github.com/kubernetes/kubernetes/blob/master/hack/local-up-cluster.sh#L245) and[ install_cni func](https://github.com/kubernetes/kubernetes/blob/master/hack/local-up-cluster.sh#L1186) is written to work seamlessly for that I believe which is why its setting up the `/etc/cni/net.d/10-containerd-net.conflist` in that function.
We can let the `local-up-cluster` default to  always call `install_cni` [here](https://github.com/kubernetes/kubernetes/blob/master/hack/local-up-cluster.sh#L1365) and strip [install_cni_if_needed](https://github.com/kubernetes/kubernetes/blob/master/hack/local-up-cluster.sh#L1365) if that's what you are pitching ?
I'm open to ideas.

Reply:
I don't have any strong opinion here. It just feels odd to test for `/etc/cni/net.d/10-containerd-net.conflist` when the real question that needs to be answered is "is cni up-to-date or do we need to re-install".

Reply:
cc @dims

also mixed feelings on this one, haven't had much time to think through what is most reasonable here, not using this much at the moment, stumbled back across this PR sweeping sig testing issues.

Reply:
let's close this out for now.

/close